metashare/api/jobs.py

```diff
@@ -71,6 +77,44 @@ def _create_branches_on_github(*, user, repo_id, project, task):
     return task_branch_name
 
 
+def alert_user_about_expiring_org(org, user):
+    from .models import User, ScratchOrg
+
+    def _do_alert():
+        # if scratch org is there
+        try:
+            org.refresh_from_db()
+            user.refresh_from_db()
+        except (ScratchOrg.DoesNotExist, User.DoesNotExist):
+            return
+
+        # and has unsaved changes
+        if org.unsaved_changes:
+            domain = Site.objects.first().domain
+            path = reverse("github_login")
+            metashare_login_link = f"https://{domain}{path}"
+            # email user
+            send_mail(
+                "MetaShare Scratch Org Expiring with Uncommitted Changes",
+                render_to_string(
+                    "scratch_org_expiry_email.txt",
+                    {
+                        "repo_name": org.task.project.repository.name,
+                        "project_name": org.task.project.name,
+                        "task_name": org.task.name,
+                        "expiry_date": org.expires_at,
+                        "user_name": user.username,
+                        "metashare_login_link": metashare_login_link,
+                    },
+                ),
+                settings.DEFAULT_FROM_EMAIL,
+                [user.email],
+                fail_silently=False,
+            )
+
+    return _do_alert
```

Comment:
Oh huh, does scheduling a closure actually work? How does it serialize that for the queue?

Reply:
I have no idea how it does; we should hand-test that to confirm that it works, maybe!

Reply:
@wlonk See my comment below -- it does not seem to be working. https://github.com/oddbird/MetaShare/pull/154#discussion_r373022174

Reply:
I saw, and changed accordingly.